File: front/Database.Migration.System/src/app/api/models/user-migration-data.ts
/* tslint:disable */
/* eslint-disable */
import { MigrationStatus } from './migration-status';
import { TimeSpan } from './time-span';
export interface UserMigrationData {
  currentRecordsCount?: number;
  migrationDuration?: TimeSpan;
  migrationId?: string;
  migrationProgressPercentage?: number;
  migrationStatus?: MigrationStatus;
  totalRecordsForMigration?: number;
}

File: DatabaseMigrationSystem.Common/Dto/UserMigrationData.cs
using DatabaseMigrationSystem.Common.Enums;

namespace DatabaseMigrationSystem.Common.Dto;

public class UserMigrationData
{
    public long TotalRecordsForMigration { get; set; }
    public long CurrentRecordsCount { get; set; }
    public MigrationStatus MigrationStatus { get; set; }
    public TimeSpan MigrationDuration { get; set; }
    public Guid MigrationId { get; set; }
    public double MigrationProgressPercentage { get; set; }
    
    public DateTime Date { get; set; }
}